<!doctype html>
<html lang="en">

<head>
	<meta charset="utf-8">
	<title>Gene interaction</title>
	<meta name="author" content="Alex Gavryushkin">
	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
	<meta name="apple-mobile-web-app-capable" content="yes" />
	<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
	<link rel="stylesheet" href="css/reveal.css">
	<link rel="stylesheet" href="css/theme/simple_gavruskin.css" id="theme">
	<link rel="stylesheet" href="lib/css/zenburn.css">
	<!--[if lt IE 9]>
	<script src="lib/js/html5shiv.js"></script>
	<![endif]-->
</head>

<body>
<div class="reveal">
<div class="slides">

<section>
	<h2>Gene interactions:<br>
	a geometric approach</h2>
	<h3>Alex Gavryushkin</h3>
	<img data-src="images/eth_logo_kurz_pos.eps.png">
	<br>
	Joint work with<br>
	Kristina Crona,	American University, Washington, DC, USA;<br>
	Bernd Sturmfels, University of California, Berkeley, USA;<br>
	Ewa Szczurek, University of Warsaw, Poland;<br>
	Niko Beerenwinkel, ETH Zurich, Basel, Switzerland
	<br>
	<br>
	October 5, 2016
</section>

<section>
	<h3>Surprised I'm not talking about trees?</h3>
	Don't worry.
	<ul>
	<li> All my talks (including this one) are here:<br>
	<a href="http://alex.gavruskin.com/talks">http://alex.gavruskin.com/talks</a><br>
	<li> Preprints are here:<br>
	<a href="http://alex.gavruskin.com/publications">http://alex.gavruskin.com/publications</a><br>
	<li> And a blog post about what I've recently been up to is here:<br>
	<a href="http://matsen.fredhutch.org/general/2016/07/11/discrete-time-tree.html">http://matsen.fredhutch.org/general/2016/07/11/discrete-time-tree.html</a>
	</ul>
</section>

<section>
	<h3>Epistasis</h3>
	<br>
	<h4>Additive allelic effect:</h4>
	$$
	w_{11} + w_{00} = w_{01} + w_{10}
	$$
	<br>
	<h4>
	<p class="fragment" data-fragment-index="1">
	Deviation from the additive expectation of allelic effects:
	<p>
	</h4>
	<p class="fragment" data-fragment-index="1">
	$u_{11} = w_{00} + w_{11} - (w_{01} + w_{10})$
	</p>
</section>

<section>
	<h3>Understanding three-way interactions</h3>

	<h4>
	<p class="fragment" data-fragment-index="1">
	Marginal epistasis?
	</p>
	</h4>
	<p class="fragment" data-fragment-index="1">
	$u_{\color{blue}{0}11} = w_{\color{blue}{0}00} + w_{\color{blue}{1}00} + w_{\color{blue}{0}11} + w_{\color{blue}{1}11} − (w_{\color{blue}{0}01} + w_{\color{blue}{1}01}) − (w_{\color{blue}{0}10} + w_{\color{blue}{1}10})$
	</p>
	
	<h4>
	<p class="fragment" data-fragment-index="2">
	Total three-way interaction?
	</p>
	</h4>
	 <p class="fragment" data-fragment-index="2">
	$u_{111} = w_{000} + w_{011} + w_{101} + w_{110} - (w_{001} + w_{010} + w_{100} + w_{111})$
	</p>

	<h4>
	<p class="fragment" data-fragment-index="3">
	Conditional epistasis?
	<p>
	</h4>
	<p class="fragment" data-fragment-index="3">
	$e = w_{\color{blue}{0}00} − w_{\color{blue}{0}01} − w_{\color{blue}{0}10} + w_{\color{blue}{0}11}$
	</p>
	<h4>
	<p class="fragment" data-fragment-index="4">
	Total mess!
	</p>
	</h4>
</section>

<section>
	<h3>(Algebraic) Geometry sorts out the mess!</h3>
	$e = u_{011} + u_{111}$
	<p class="fragment" data-fragment-index="1">
	In general, the four <i>interaction coordinates</i>
	$$
	u_{011}, u_{101}, u_{110}, u_{111}
	$$
	allow to describe <i>all</i> possible kinds of interaction!
	</p>
	<p class="fragment" data-fragment-index="2">
	There are 20 types of interaction and they are known as <i>circuits</i> to Algebraic Geometry 111 students
	</p>
	<p class="fragment" data-fragment-index="3">
	Yep, we've got the list!
	</p>
</section>

<section data-transition="slide-in none-out">
	$$
	\begin{align*}
	a&= w_{000}-w_{010}-w_{100}+w_{110} & m&=w_{001}+w_{010}+w_{100}-w_{111}-2w_{000}\\
	b&=w_{001}-w_{011}-w_{101}+w_{111} & n&=w_{011}+w_{101}+w_{110}-w_{000}-2w_{111}\\
	c&=w_{000}-w_{001}-w_{100}+w_{101} & o&=w_{010}+w_{100}+w_{111}-w_{001}-2w_{110}\\
	d&=w_{010}-w_{011}-w_{110}+w_{111} & p&=w_{000}+w_{011}+w_{101}-w_{110}-2w_{001}\\
	e&=w_{000}-w_{001}-w_{010}+w_{011} & q&=w_{001}+w_{100}+ w_{111}-w_{010}-2w_{101}\\
	f&=w_{100}-w_{101}-w_{110}+w_{111} & r&=w_{000}+w_{011}+ w_{110}-w_{101}-2w_{010}\\
	g&=w_{000}-w_{011}-w_{100}+w_{111} & s&=w_{000}+w_{101}+ w_{110}-w_{011}-2w_{100}\\
	h&=w_{001}-w_{010}-w_{101}+w_{110} & t&=w_{001}+w_{010}+w_{111}-w_{100}-2w_{011}\\
	i&=w_{000}-w_{010}-w_{101}+w_{111}\\
	j&=w_{001}-w_{011}-w_{100}+w_{110}\\
	k&=w_{000}-w_{001}-w_{110}+w_{111}\\
	l&=w_{010}-w_{011}-w_{100}+w_{101}\\
	\end{align*}
	$$
</section>

<section data-transition="none-in slide-out">
	$$
	\begin{align*}
	a&= w_{000}-w_{010}-w_{100}+w_{110} & m&=w_{001}+w_{010}+w_{100}-w_{111}-2w_{000}\\
	b&=w_{001}-w_{011}-w_{101}+w_{111} & n&=w_{011}+w_{101}+w_{110}-w_{000}-2w_{111}\\
	c&=w_{000}-w_{001}-w_{100}+w_{101} & o&=w_{010}+w_{100}+w_{111}-w_{001}-2w_{110}\\
	d&=w_{010}-w_{011}-w_{110}+w_{111} & p&=w_{000}+w_{011}+w_{101}-w_{110}-2w_{001}\\
	e&=w_{000}-w_{001}-w_{010}+w_{011} & q&=w_{001}+w_{100}+ w_{111}-w_{010}-2w_{101}\\
	f&=w_{100}-w_{101}-w_{110}+w_{111} & r&=w_{000}+w_{011}+ w_{110}-w_{101}-2w_{010}\\
	\color{blue}{g}&\color{blue}{=w_{000}-w_{011}-w_{100}+w_{111}} & s&=w_{000}+w_{101}+ w_{110}-w_{011}-2w_{100}\\
	h&=w_{001}-w_{010}-w_{101}+w_{110} & t&=w_{001}+w_{010}+w_{111}-w_{100}-2w_{011}\\
	i&=w_{000}-w_{010}-w_{101}+w_{111}\\
	j&=w_{001}-w_{011}-w_{100}+w_{110}\\
	k&=w_{000}-w_{001}-w_{110}+w_{111}\\
	l&=w_{010}-w_{011}-w_{100}+w_{101}\\
	\end{align*}
	$$
</section>

<section>
	<h3>This is known as Beerenwinkel-Pachter-Sturmfels approach,</h3>
	which provides a complete picture of interactions!<br>
	<br>
	<h3>BUT</h3>
	<p class="fragment" data-fragment-index="1">
	the approach is
	</p>
	<ul>
	<li>
	<p class="fragment" data-fragment-index="1">
	based on the availability of <b>all</b> fitness measurements!
	</p>
	<li>
	<p class="fragment" data-fragment-index="1">
	computationally feasible for up to four loci!
	</p>
	</ul>
	<h3>
	<p class="fragment" data-fragment-index="2">
	Hence, we come to two research questions
	</p>
	</h3>
</section>

<section>
	<h3><u>Problem 1:</u> What if no credible fitness measurements are available?</h3>
	<p class="fragment" data-fragment-index="1">
	Like in this malaria drug resistance data set:
	<img class=stretch data-src="images/tem.svg">
	<br>
	Ogbunugafor et al. <i>Malar. J.</i> 2016
	</p>
</section>

<section>
	<h3>Results at a glance</h3>
	<ul>
	<li> We provide a complete characterization of <i>fitness graphs</i> that imply circuit interaction (think epistasis).
	<li> Fitness graphs arise in competition-like experiments and include:
		<ul>
		<li> Rank orders
		<li> Mutation graphs
		</ul>
	</ul>
	<br>
	<br>
	(Preprints with Crona, Beerenwinkel, and others to appear)
</section>

<section>
	<h4>Rank orders. The simplest case.</h4>
	<img class=stretch data-src="images/all_24_rankings_two_loci.svg">
</section>

<section>
	<h4>Mutation graph</h4>
	<img class=stretch data-src="images/tem.svg">
</section>

<section>
	<h4>Connection between the two</h4>
	<img class=stretch data-src="images/all_total_extensions_of_p_o.svg">
</section>

<section>
	<h3>Applications</h3>
	<ul>
	<li> HIV-1
	<li> Antibiotic resistance
	<li> Fungi
	<li> Gut microbiome (with Will Ludington, UC Berkeley)
	<li> Synthetic lethality
	<li> Knockdown cell lines
	</ul>
	<br>
	<br>
	<p class="fragment" data-fragment-index="1">
	Methodologically, this allows us to advise further measurements (experiments) for incomplete data sets. This exponentially reduce the number of potential experiments (see HIV example).
	</p>
</section>

<section>
	<h3>Results in more detail</h3>
	Efficient methods for:
	<ul>
	<li> Circuit interaction inference (including epistasis and three-way interaction) for total orders
	<li> Complete analysis of partial orders (including mutation graphs) with "distance to interaction" inference
	<li> Suggestions for possible completion in case of missing data and/or high uncertainty
	</ul>
	<br>
	<br>
	Software (pre-release stage):<br>
	<a href="https://github.com/gavruskin/fitlands">https://github.com/gavruskin/fitlands</a>
</section>

<section>
	<h3><u>Problem 2:</u> What if the number of genes (loci) is 100?</h3>
	<ul>
	<li>2^98 of conditional epistases?
	<li>2^98 measurements to estimate marginal epistasis?
	</ul>
	<p>Not in this life</p>
	<h3>
	<p class="fragment" data-fragment-index="1">
	Concrete example: genome-wide RNAi perturbation screens
	</p>
	</h3>
	<p class="fragment" data-fragment-index="1">
	100 genes, 1000 experiments
	</p>
</section>

<section>
	<h3>Two ways out</h3>
	<ol>
	<li> Ask biologists about "the most important genes" that are main fitness drivers (like we did in the HIV study)
	<li> Add statistical assumptions, for example:
		<ul>
		<li> Ignore higher-order terms in Taylor series approximations
		<li> Structural hypotheses
		</ul>
	</ol>
	<br>
	<br>
	(Preprints with Szczurek, Beerenwinkel, et al. to appear)
</section>

<section>
	<h3>
	Thanks for your attention!<br>
	And stay tuned!
	</h3>
	<br>
	<br>
	Funding:
	<br>
	<img data-src="images/cbg_ethz.svg">
</section>
</div>
</div>
<script src="lib/js/head.min.js"></script>
<script src="js/reveal.js"></script>
<script>
  Reveal.initialize({
    dependencies: [
        // Interpret Markdown in <section> elements
        { src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
        { src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
        // MathJax
        { src: 'plugin/math/math.js', async: true }
    ]
  });
</script>
</body>
</html>

