---
layout: post
title:  "Postdoc with us!"
date:   2018-03-26 12:00:00 +1300
categories: jobs
---

We are thrilled to announce that we will soon start a formal search for new lab members!
These will include all levels, from interns to postdocs, but in this post we wish to talk about how to become the inaugural postdoc at the [Biological Data Science lab](http://lab.gavruskin.com) within the [Department of Computer Science](http://www.cs.otago.ac.nz/), [University of Otago](http://www.otago.ac.nz/).

On the practical side of things, we will be offering a one year fixed-term position in the first instance, but we are also quite flexible and the final offer will be tailored to the successful applicant.
The appointment will be made at the Postdoctoral Fellow or Research Fellow [level](http://www.otago.ac.nz/humanresources/working-at-otago/salaries-and-payments/pay-scales/index.html), depending on experience.
The successful candidate will have a PhD in mathematics, computer science, statistics, bioinformatics, or a related area.
The work will be carried out in beautiful Dunedin, the gateway to New Zealand’s most magnificent region of Otago.

Now, the interesting part.
At our lab, you will be presented with a unique opportunity to chose from a variety of interdisciplinary projects the one (or more!) that suites you best, build and expand your network of collaborators, and move your science to the next level.
Read on to learn about things that we find exciting at our lab and feel free to suggest more!

Evolutionary analysis of large molecular sequence data is widely employed throughout modern biology, medicine, and pharmacology.
Thanks to next generation sequencing technologies, thousands of whole genome sequences are constantly produced at low cost.
Today’s computational methods and technologies are barely prepared to analyse big omics data and answer basic biological questions in a statistically sound way.
As a result many big data sets are analysed with simple and inappropriate models or using approximations that may produce inaccurate inferences.
The roots of the problem lie in our lack of understanding of fundamental mathematical principles that form the grounds for evolutionary analysis of omics data.
Many recent breakthroughs in computational biology were made possible due to mathematical advances.
Previously unimagined fields of applied mathematics such as mathematical oncology are currently being developed to help clinicians make diagnostic and treatment decisions based on quantitatively justified verifiable grounds.
More and more healthcare institutions are hiring mathematicians and computer scientists to deal with the problems arising in these areas.
The pressing need to develop effective computational methods that can accurately analyse big omics data under appropriately complex evolutionary models constantly requires new algorithmic ideas as well as solutions of standing computational challenges.
The ever-accelerating pace at which genomic, transcriptomic, and proteomic data is produced in the modern world makes it impractical to rerun all analyses every time new data arrives or existing data is refined.
The need to rerun the same analysis over and over again with slightly different data is a fundamental reason why inaccurate and often unreliable heuristics are favoured over statistically sound methods.

This motivates research on so-called "online" computational biology algorithms capable of integrating new data as it appears.
At our lab, you will be presented with is a unique opportunity to contribute to the rapidly developing field of online computational biology at its early stages.
As a successful candidate, you will take up a leading role in the development of online algorithms for evolutionary analysis of omics data.
You will apply methods from modern branches of computational geometry, data science, or computer science to enhance statistical and computational performance of evolutionary approaches used in genomics, cancer research, ecology, or pharmacology.

Since the vast majority of algorithmic problems in computational biology currently lack online solutions, this project would suit a diverse range of experts in computer science, mathematics, statistics, and quantitative biology or biomedicine.
The examples of immediate interest to our lab include, but are absolutely not limited to: phylogenetic tree inference algorithms (from distance-based to full Bayesian methods), algorithms used in fitness landscape studies, gene interactions inference, and other areas of computational biology where greedy algorithms play a major role.

This project is embedded in the highly interdisciplinary Rutherford Discovery Fellowship research programme led by [Dr Alex Gavryushkin](http://lab.gavruskin.com/alex/).
Our collaborators span various divisions at Otago (Mathematics & Statistic, Biochemistry, Dunedin School of Medicine), universities in New Zealand (Auckland and Canterbury) and overseas (ETH Zurich, Max Planck Institutes in Plön and Leipzig, Fred Hutch Cancer Research Centre in Seattle).
An opportunity to collaborate with these institutions will be available to the successful candidate.

Find this interesting? So do we! Get in touch with [Alex](mailto:alex@gavruskin.com), who is keen to hear from you.
